### PR TITLE
[GBM] Unbreak on FreeBSD

### PR DESCRIPTION
**Pull request by ghost**
## Description

- Link with libepoll-shim [1] on FreeBSD. FreeBSD does not have epoll but there is a kqueue-based shim for it called epoll-shim. This allows the GBM platform to build and link properly on FreeBSD.
- Use EPOLL_CLOEXEC for epoll1_create for now due to lack of features in epoll-shim [2]. It is not clear if this is appropriate for Linux or not, so I put this behind TARGET_FREEBSD for now.

[1] https://github.com/jiixyj/epoll-shim/
[2] https://github.com/jiixyj/epoll-shim/blob/9185474a4b382823cfaeaeaabfa75b277890c9e7/src/epoll.c#L97

## How Has This Been Tested?

Build and run tested on FreeBSD. The patches are part of the kodi-devel package on FreeBSD.

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/platform/linux/input/LibInputHandler.cpp [hunk 1]
```diff
@@ -116,7 +116,14 @@ void CLibInputHandler::Start()
 
 void CLibInputHandler::Process()
 {
+#if defined(TARGET_FREEBSD)
+  // epoll_create1() from libepoll-shim always fails with 0 and
+  // only supports EPOLL_CLOEXEC, so let's use that to make Kodi/GBM
+  // runnable on FreeBSD.
+  int epollFd = epoll_create1(EPOLL_CLOEXEC);
```
Comment (lrusak) [inline xbmc/platform/linux/input/LibInputHandler.cpp]: Can you just make this the default and remove the ifdeffing? I'll give it a go on linux but I don't see why it wouldn't work.

Comment (t6) [reply xbmc/platform/linux/input/LibInputHandler.cpp]: Ok, done.

Comment (t6) [reply xbmc/platform/linux/input/LibInputHandler.cpp]: So was it fine with EPOLL_CLOEXEC on Linux too?

Comment (lrusak) [reply xbmc/platform/linux/input/LibInputHandler.cpp]: Seems fine to me. I just tested it. Sorry for the delay. I just moved houses and finally set my workstation up again.

**Review by lrusak — APPROVED** [review]

**Comment by lrusak** [comment]
jenkins build this please

**Comment by t6** [comment]
Thanks for testing and approving, @lrusak. What needs to happen next to get it merged?

**Comment by fritsch** [comment]
jenkins build this please (something went wrong)

**Comment by t6** [comment]
@fritsch It seems like something went wrong again with Jenkins?

**Comment by fritsch** [comment]
jenkins build this please

**Comment by t6** [comment]
Looks like the build was fine.

**Comment by t6** [comment]
Ping?

**Comment by lrusak** [comment]
jenkins build this please

**Comment by lrusak** [comment]
Thanks for fixing this and sorry it took so long to merge! Hopefully in the future we can be more accommodating of FreeBSD changes now that our jenkins is building all green again 👍

**Pull request merged**